src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -388,6 +401,15 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	}
 
 	public async requestConnection(): Promise<boolean> {
+		// If there is a saved connection name with a corresponding connection profile, use that one,
+		// otherwise show connection dialog
+		if (this.configurationService.getValue(saveConnectionNameConfigName) && this._savedConnectionName) {
+			let profile: ConnectionProfile | undefined = this.getConnectionProfileFromName(this._savedConnectionName);
+			if (profile) {
+				await this.changeContext(this._savedConnectionName, profile);
+				return true;
+			} // TODO: No matching connection profile for saved connection name
```

Comment:
Is this TODO still needed?

Reply:
I can remove this comment for now since we haven't fully decided whether we want to go with the alias or data source name approach yet.